ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java

```diff
@@ -462,6 +462,41 @@ public void shouldHandleMaps() {
         assertThat(result, is("value1"));
     }
 
+    @Test
+    public void shouldHandleCaseStatement() {
+        // Given:
+        final Expression expression = analyzeQuery(
+            "SELECT CASE WHEN col0 < 10 THEN 'small' WHEN col0 < 100 THEN 'medium' ELSE 'large' END FROM codegen_test;", metaStore)
+            .getSelectExpressions()
+            .get(0);
+
+        // When:
+        final Object result = codeGenRunner
+            .buildCodeGenFromParseTree(expression, "Case")
+            .evaluate(genericRow(ONE_ROW));
+
+        // Then:
+        assertThat(result, is("small"));
+    }
+
+    @Test
+    public void shouldReturnDefaultForCaseCorrectly() {
```

Comment:
Add test case that has no `ELSE` statement and where none of the `WHEN` match, confirming the result is `null`